### PR TITLE
avoid c-ares dependency on libnsl (for v1.17.x)

### PR DESCRIPTION
**Pull request by jtattermusch**
Fix for #17255.

Also see https://github.com/grpc/grpc/pull/17270

Still needs to be upmerged to master.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,496      Total (=)      2,020,496

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,175,628      Total (>)     11,175,627

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by jtattermusch** [comment]
This nuget with this change passes with a makeshift C# distribtest on alpine: https://source.cloud.google.com/results/invocations/f3c0d70f-066f-4dee-8d17-cca44bf26194/targets/github%2Fgrpc/tests;query=alpine;failed=false;passed=true

**Comment by apolcyn** [comment]
Saw https://github.com/grpc/grpc/pull/17272, but just for the idea I'm wondering if a set of C tests that run under `BACKWARDS_COMPATIBILITY_MODE` would be useful

File: cmake/cares.cmake [hunk 1]
```diff
@@ -18,6 +18,10 @@ if("${gRPC_CARES_PROVIDER}" STREQUAL "module")
   endif()
   set(CARES_SHARED OFF CACHE BOOL "disable shared library")
   set(CARES_STATIC ON CACHE BOOL "link cares statically")
+  if(gRPC_BACKWARDS_COMPATIBILITY_MODE)
+    # See https://github.com/grpc/grpc/issues/17255
```
Comment (apolcyn) [inline cmake/cares.cmake]: can we add a comment here that we expect the C# binary packages to be the only builds that use this?

Comment (jtattermusch) [reply cmake/cares.cmake]: I think the issue link already hints that this is related to C# packaging.

**Pull request merged**